shell/profiling/sampling_profiler.cc

```diff
@@ -16,7 +16,13 @@ SamplingProfiler::SamplingProfiler(
       sampler_(std::move(sampler)),
       num_samples_per_sec_(num_samples_per_sec) {}
 
-void SamplingProfiler::Start() const {
+SamplingProfiler::~SamplingProfiler() {
+  if (is_running_) {
+    Stop();
```

Comment:
Can we add some tests? i.e. you can start and stop and start etc and check the scheduling of samples (even if you mock out the meaty parts). And check that destroying indeed stops.

Reply:
Man, what a doozy to test, specially since there wasn't any existing tests.  Ok, added.